This data analyzes the PPP data from https://data.sba.gov/dataset/ppp-foia/resource/6dd520ff-c211-404f-b7eb-29784cd4dbda?inner_span=True

In [2]:
import pandas as pd
pd.options.display.max_colwidth = 400
pd.options.display.max_columns = 90

In [4]:
df = pd.read_csv('public_up_to_150k_1_220930.csv')

In [5]:
df.head()

LoanNumber DateApproved  SBAOfficeCode ProcessingMethod  \
0  3509338307   01/22/2021            NaN              PPS   
1  5375617707   05/01/2020          101.0              PPP   
2  9677497701   05/01/2020          464.0              PPP   
3  9547167709   05/01/2020          464.0              PPP   
4  8885207205   04/28/2020            NaN              PPP   

                             BorrowerName          BorrowerAddress  \
0                             Exemption 6                      NaN   
1                           NOT AVAILABLE                      NaN   
2  NORTH CHARLESTON HOSPITALITY GROUP LLC      192 College Park Rd   
3                    Q AND J SERVICES LLC  301 Old Georgetown Road   
4                             Exemption 6                      NaN   

  BorrowerCity BorrowerState BorrowerZip LoanStatusDate    LoanStatus  Term  \
0          NaN           NaN         NaN     02/18/2022  Paid in Full    60   
1          NaN           NaN         NaN     07/16/2021  Paid in Full    24   
2       Ladson           NaN  29456-3517     09/25/2021  Paid in Full    24   
3      Manning           NaN  29102-2734     04/20/2021  Paid in Full    24   
4          NaN           NaN         NaN     05/20/2021  Paid in Full    24   

   SBAGuarantyPercentage  InitialApprovalAmount  CurrentApprovalAmount  \
0                    100              149360.60              149360.60   
1                    100              148440.00              148440.00   
2                    100              140081.78              140081.78   
3                    100              136520.96              136520.96   
4                    100              130600.00              130600.00   

   UndisbursedAmount FranchiseName  ServicingLenderLocationID  \
0                0.0           NaN                        NaN   
1                0.0           NaN                     9551.0   
2                0.0           NaN                    19248.0   
3                0.0           NaN                    19248.0   
4                0.0           NaN                        NaN   

                     ServicingLenderName   ServicingLenderAddress  \
0                                    NaN                      NaN   
1  Bank of America, National Association  100 N Tryon St, Ste 170   
2                           Synovus Bank            1148 Broadway   
3                           Synovus Bank            1148 Broadway   
4                                    NaN                      NaN   

  ServicingLenderCity ServicingLenderState ServicingLenderZip  \
0                 NaN                  NaN                NaN   
1           CHARLOTTE                   NC         28202-4024   
2            COLUMBUS                   GA         31901-2429   
3            COLUMBUS                   GA         31901-2429   
4                 NaN                  NaN                NaN   

  RuralUrbanIndicator HubzoneIndicator LMIIndicator  \
0                   U                N            N   
1                   U                N            N   
2                   U                N            N   
3                   U                Y            Y   
4                   U                N            N   

              BusinessAgeDescription ProjectCity ProjectCountyName  \
0  Existing or more than 2 years old         NaN               NaN   
1    New Business or 2 years or less         NaN               NaN   
2  Existing or more than 2 years old      Ladson          BERKELEY   
3  Existing or more than 2 years old     Manning         CLARENDON   
4                         Unanswered         NaN               NaN   

  ProjectState  ProjectZip     CD  JobsReported  NAICSCode        Race  \
0          NaN         NaN    NaN          15.0   624190.0  Unanswered   
1          NaN         NaN    NaN          12.0   339114.0  Unanswered   
2           SC  29456-3517  SC-01           3.0        NaN  Unanswered   
3           SC  29102-2734  SC-06         17

In [6]:
df.shape

(900000, 53)

The first section of this table is 900K records.

In [5]:
df.columns

Index(['LoanNumber', 'DateApproved', 'SBAOfficeCode', 'ProcessingMethod',
       'BorrowerName', 'BorrowerAddress', 'BorrowerCity', 'BorrowerState',
       'BorrowerZip', 'LoanStatusDate', 'LoanStatus', 'Term',
       'SBAGuarantyPercentage', 'InitialApprovalAmount',
       'CurrentApprovalAmount', 'UndisbursedAmount', 'FranchiseName',
       'ServicingLenderLocationID', 'ServicingLenderName',
       'ServicingLenderAddress', 'ServicingLenderCity', 'ServicingLenderState',
       'ServicingLenderZip', 'RuralUrbanIndicator', 'HubzoneIndicator',
       'LMIIndicator', 'BusinessAgeDescription', 'ProjectCity',
       'ProjectCountyName', 'ProjectState', 'ProjectZip', 'CD', 'JobsReported',
       'NAICSCode', 'Race', 'Ethnicity', 'UTILITIES_PROCEED',
       'PAYROLL_PROCEED', 'MORTGAGE_INTEREST_PROCEED', 'RENT_PROCEED',
       'REFINANCE_EIDL_PROCEED', 'HEALTH_CARE_PROCEED',
       'DEBT_INTEREST_PROCEED', 'BusinessType', 'OriginatingLenderLocationID',
       'OriginatingLender', 'Originatin

In [6]:
df['Race'].value_counts(normalize=True)

Unanswered                                   0.753040
White                                        0.132942
Asian                                        0.050910
Black or African American                    0.049590
American Indian or Alaska Native             0.012229
Native Hawaiian or Other Pacific Islander    0.001262
Multi Group                                  0.000017
Eskimo & Aleut                               0.000009
Puerto Rican                                 0.000001
Name: Race, dtype: float64

In [84]:
df['Ethnicity'].value_counts()

Unknown/NotStated         637661
Not Hispanic or Latino    232552
Hispanic or Latino         29785
Name: Ethnicity, dtype: int64

Of the 900K records, 75% do not indicate race.

What percent of these records are churches (or other religious institutions?)

of the various payment categories, how many are used in our DB of 900K records?

In [89]:
PayCats = ['UTILITIES_PROCEED','PAYROLL_PROCEED','MORTGAGE_INTEREST_PROCEED','RENT_PROCEED','REFINANCE_EIDL_PROCEED','HEALTH_CARE_PROCEED','DEBT_INTEREST_PROCEED']
for cat in PayCats:
    print(f'Category {cat} has {df[cat].isna().sum()} blank rows or {df[cat].isna().sum() / 900000 * 100:.2f} percent of the total.')



Category UTILITIES_PROCEED has 628170 blank rows or 69.80 percent of the total.
Category PAYROLL_PROCEED has 933 blank rows or 0.10 percent of the total.
Category MORTGAGE_INTEREST_PROCEED has 880676 blank rows or 97.85 percent of the total.
Category RENT_PROCEED has 850597 blank rows or 94.51 percent of the total.
Category REFINANCE_EIDL_PROCEED has 891506 blank rows or 99.06 percent of the total.
Category HEALTH_CARE_PROCEED has 880932 blank rows or 97.88 percent of the total.
Category DEBT_INTEREST_PROCEED has 888530 blank rows or 98.73 percent of the total.


In [25]:
df[df['BorrowerName'].str.contains('CHURCH | TEMPLE | MOSQUE | MASJID | SYNAGOGUE | GURUDWARA')][['BorrowerName','CurrentApprovalAmount','ForgivenessAmount','UTILITIES_PROCEED','PAYROLL_PROCEED','NAICSCode']]

BorrowerName  \
697           RABBIT CREEK COMMUNITY CHURCH OF ANCHORAGE, ALASKA   
3373               WASILLA CHRISTIAN CHURCH DBA CREEKSIDE CHURCH   
3560                                      SOLDOTNA CHURCH OF GOD   
3659                                      SOLDOTNA CHURCH OF GOD   
4205                  THE FIRST LUTHERAN CHURCH OF KETCHIKAN, AK   
...                                                          ...   
837965                           THE EPISCOPAL CHURCH IN ALMADEN   
838738  ST. PETER'S EPISCOPAL CHURCH IN REDWOOD CITY, CALIFORNIA   
865899                                           GODS CHURCH LLC   
895594                                         DERRICK TEMPLE JR   
899199                  THE JOURNEY A CHURCH FOR THE COMMUNITY O   

        CurrentApprovalAmount  ForgivenessAmount  UTILITIES_PROCEED  \
697                 127730.00          116835.96                NaN   
3373                 68162.00           68720.37                NaN   
3560                 65510.00           66050.23                1.0   
3659                 64238.32           64829.66                NaN   
4205                 58059.00           58435.99                NaN   
...                       ...                ...                ...   
837965               20886.00           21122.33                NaN   
838738               20863.92           21084.15                NaN   
865899               20833.00           21120.67                NaN   
895594               20833.00                NaN                NaN   
899199               20833.00           20971.70                NaN   

        PAYROLL_PROCEED  NAICSCode  
697           127730.00   813110.0  
3373           68162.00   813110.0  
3560           65508.00   813110.0  
3659           49988.16   813110.0  
4205           58059.00   813110.0  
...                 ...        ...  
837965         20886.00   813110.0  
838738         20863.92   813110.0  
865899         20832.00   711510.0  
895594         20833.00   541320.0  
899199         20832.00   813110.0  

[2638 rows x 6 columns]

In [35]:
2638/900000*100

0.2931111111111111

In [24]:
# 813110 is the NAICS code for religious organizations -- https://www.census.gov/naics/?input=813110&year=2022&details=813110
# "This industry comprises (1) establishments primarily engaged in operating religious organizations, such as churches, religious temples, mosques, and monasteries, and/or 
# (2) establishments primarily engaged in administering an organized religion or promoting religious activities."

# this is probably the best / most inclusive way to identify religious organizations in the PPP data. (This came up with 9620 records, whereas my search for religious language in names only 
# found 2638!)
df[df['NAICSCode'] == 813110]

LoanNumber DateApproved  SBAOfficeCode ProcessingMethod  \
32      9772027709   05/01/2020          101.0              PPP   
113     9727847709   05/01/2020          101.0              PPP   
198     5175837208   04/27/2020         1084.0              PPP   
300     2355348508   02/20/2021         1084.0              PPS   
552     2179217203   04/15/2020         1084.0              PPP   
...            ...          ...            ...              ...   
877932  4667838805   04/16/2021          914.0              PPS   
879981  4895588706   04/01/2021          912.0              PPP   
881530  5085457807   05/29/2020          912.0              PPP   
893189  6186478410   02/10/2021          912.0              PPS   
899199  6663958405   02/10/2021          920.0              PPS   

                                    BorrowerName      BorrowerAddress  \
32                   HOLY TRINITY BAPTIST CHURCH                  NaN   
113              RHEMA EMPOWERMENT CENTER CHURCH                  NaN   
198                     COVENANT YOUTH OF ALASKA       228 W 11th Ave   
300             ST. JOHN UNITED METHODIST CHURCH      1801 Omalley Rd   
552              LIGHTHOUSE CHRISTIAN FELLOWSHIP  615 HOLLYWOOD DRIVE   
...                                          ...                  ...   
877932                                 LEVI WADE    212 w 73rd street   
879981                          GENEIVEVE PIERTE    1323 Steinbeck Dr   
881530                              EQUIPNET INC   1510 HILLGRADE AVE   
893189                               AHRON HECHT         423 10th Ave   
899199  THE JOURNEY A CHURCH FOR THE COMMUNITY O    23655 Via del Rio   

         BorrowerCity BorrowerState BorrowerZip LoanStatusDate    LoanStatus  \
32                NaN           NaN         NaN     11/17/2021  Paid in Full   
113               NaN           NaN         NaN     08/17/2021  Paid in Full   
198         Anchorage            AK       99501     04/21/2021  Paid in Full   
300         Anchorage            AK  99507-7304     01/06/2022  Paid in Full   
552         ANCHORAGE            AK       99501     05/12/2021  Paid in Full   
...               ...           ...         ...            ...           ...   
877932    Los Angeles            CA       90003            NaN   Exemption 4   
879981      Pittsburg            CA  94565-3665     10/22/2021  Paid in Full   
881530          ALAMO            CA  94507-2603     06/10/2021  Paid in Full   
893189  San Francisco            CA  94118-2914     10/16/2021  Paid in Full   
899199    Yorba Linda            CA  92887-2718     11/19/2021  Paid in Full   

        Term  SBAGuarantyPercentage  InitialApprovalAmount  \
32        24                    100               32625.00   
113       24                    100                9000.00   
198       24                    100              148032.50   
300       60                    100              143092.00   
552       24                    100              133821.53   
...      ...                    ...                    ...   
877932    60                    100               20833.00   
879981    60                    100               20833.00   
881530    24                    100               20833.00   
893189    60                    100               20833.00   
899199    60                    100               20833.00   

        CurrentApprovalAmount  UndisbursedAmount FranchiseName  \
32                   32625.00                0.0           NaN   
113                   9000.00                0.0           NaN   
198                 148032.50                0.0           NaN   
300                 143092.00                0.0           NaN   
552                 133821.53                0.0           NaN   
...                       ...                ...           ...   
877932               20833.00                0.0           NaN   
879981               20833.00                0.0           NaN   
881530               20833

So 0.29% of all the PPP loans were disbursed to religious organizations.

Is there a pattern in what sorts of businesses identifed a race?

In [123]:
df['BusinessType'].value_counts(normalize=True)

Corporation                            0.298651
Sole Proprietorship                    0.257966
Limited  Liability Company(LLC)        0.182752
Subchapter S Corporation               0.107975
Self-Employed Individuals              0.055855
Independent Contractors                0.044581
Non-Profit Organization                0.020716
Partnership                            0.013698
Limited Liability Partnership          0.006135
Single Member LLC                      0.004422
Professional Association               0.003826
Cooperative                            0.001087
501(c)3 – Non Profit                   0.000949
501(c)6 – Non Profit Membership        0.000469
Trust                                  0.000273
Non-Profit Childcare Center            0.000264
Tenant in Common                       0.000129
Qualified Joint-Venture (spouses)      0.000093
Joint Venture                          0.000088
Employee Stock Ownership Plan(ESOP)    0.000043
Tribal Concerns                        0

There are wide array of business types, many have very few respondants.
Corporations (29.8%), Sole Proprietorship (25.8%), LLC (18.3%), and Subchapter S (10.8%) are 85% of the total.

Is there a pattern for those that reported entered a value for "Race"?

In [92]:
noRace = df[df['Race'] == 'Unanswered']['BusinessType'].value_counts().sort_index()

In [93]:
yesRace = df[df['Race'] != 'Unanswered']['BusinessType'].value_counts().sort_index()

In [113]:
BusTypePctRace = pd.concat([noRace,yesRace],axis=1).reset_index()
BusTypePctRace

index  BusinessType  BusinessType
0     501(c) – Non Profit except 3,4,6,             4           1.0
1        501(c)19 – Non Profit Veterans             5           1.0
2                  501(c)3 – Non Profit           713         141.0
3       501(c)6 – Non Profit Membership           321         101.0
4                           Cooperative           714         264.0
5                           Corporation        199031       69709.0
6   Employee Stock Ownership Plan(ESOP)            24          15.0
7               Independent Contractors         27513       12603.0
8                         Joint Venture            62          17.0
9       Limited  Liability Company(LLC)        124090       40359.0
10        Limited Liability Partnership          4311        1210.0
11          Non-Profit Childcare Center           197          41.0
12              Non-Profit Organization         16413        2228.0
13                          Partnership          9916        2410.0
14             Professional Association          2878         565.0
15    Qualified Joint-Venture (spouses)            75           9.0
16  Rollover as Business Start-Ups (ROB             2           NaN
17            Self-Employed Individuals         43119        7142.0
18                    Single Member LLC          3241         738.0
19                  Sole Proprietorship        168956       63174.0
20             Subchapter S Corporation         75681       21480.0
21                     Tenant in Common           109           7.0
22                      Tribal Concerns             5           5.0
23                                Trust           205          41.0

In [114]:
BusTypePctRace.columns.values[1] = 'NoRaceRpt'
BusTypePctRace.columns.values[2] = 'RaceRpt'
BusTypePctRace.columns.values[0] = 'BusType'

#BusTypePctRace.rename(columns = {BusTypePctRace.columns[1]:'NoRaceRpt', 
#                                 BusTypePctRace.columns[2]:'RaceRpt',
#                                'index':'BusType'},inplace=True)

In [115]:
BusTypePctRace.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 24 entries, 0 to 23
Data columns (total 3 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   BusType    24 non-null     object 
 1   NoRaceRpt  24 non-null     int64  
 2   RaceRpt    23 non-null     float64
dtypes: float64(1), int64(1), object(1)
memory usage: 704.0+ bytes


In [118]:
BusTypePctRace['PctNoRace'] = BusTypePctRace['NoRaceRpt'] / 9000

In [119]:
BusTypePctRace['PctRace'] = BusTypePctRace['RaceRpt']/9000

In [120]:
BusTypePctRace

BusType  NoRaceRpt  RaceRpt  PctNoRace  \
0     501(c) – Non Profit except 3,4,6,          4      1.0   0.000444   
1        501(c)19 – Non Profit Veterans          5      1.0   0.000556   
2                  501(c)3 – Non Profit        713    141.0   0.079222   
3       501(c)6 – Non Profit Membership        321    101.0   0.035667   
4                           Cooperative        714    264.0   0.079333   
5                           Corporation     199031  69709.0  22.114556   
6   Employee Stock Ownership Plan(ESOP)         24     15.0   0.002667   
7               Independent Contractors      27513  12603.0   3.057000   
8                         Joint Venture         62     17.0   0.006889   
9       Limited  Liability Company(LLC)     124090  40359.0  13.787778   
10        Limited Liability Partnership       4311   1210.0   0.479000   
11          Non-Profit Childcare Center        197     41.0   0.021889   
12              Non-Profit Organization      16413   2228.0   1.823667   
13                          Partnership       9916   2410.0   1.101778   
14             Professional Association       2878    565.0   0.319778   
15    Qualified Joint-Venture (spouses)         75      9.0   0.008333   
16  Rollover as Business Start-Ups (ROB          2      NaN   0.000222   
17            Self-Employed Individuals      43119   7142.0   4.791000   
18                    Single Member LLC       3241    738.0   0.360111   
19                  Sole Proprietorship     168956  63174.0  18.772889   
20             Subchapter S Corporation      75681  21480.0   8.409000   
21                     Tenant in Common        109      7.0   0.012111   
22                      Tribal Concerns          5      5.0   0.000556   
23                                Trust        205     41.0   0.022778   

     PctRace  
0   0.000111  
1   0.000111  
2   0.015667  
3   0.011222  
4   0.029333  
5   7.745444  
6   0.001667  
7   1.400333  
8   0.001889  
9   4.484333  
10  0.134444  
11  0.004556  
12  0.247556  
13  0.267778  
14  0.062778  
15  0.001000  
16       NaN  
17  0.793556  
18  0.082000  
19  7.019333  
20  2.386667  
21  0.000778  
22  0.000556  
23  0.004556

Of the four major categories, there are significant differences in the percent of all records that reported race:

Type                NoRace Rep      Race Rpt
Corporation             22.1%       7.74
Sole Prop               18.77       7.02
LLC                     13.79       4.48
Subchapter S            8.41        2.39

Overall, the majority of different business types didn't report a racial category.
Is this true for Black Churches (BCS)?

We will start with looking for AME / AMEZ churches as a test case.

In [22]:
df[df['BorrowerName'].str.contains('AFRICAN METHODIST')][['Race','BorrowerName','BusinessType']]

ValueError: Cannot mask with non-boolean array containing NA / NaN values

How many null (NAN) values do we have? If small, we will just drop them:

In [23]:
df['BorrowerName'].isna().sum()

2

In [7]:
df = df.dropna(subset = ['BorrowerName'])

In [25]:
df[df['BorrowerName'].str.contains('AFRICAN METHODIST')][['Race','BorrowerName','BusinessType']]

Race  \
19967                  Unanswered   
42228                  Unanswered   
142067                 Unanswered   
244018                 Unanswered   
249603                 Unanswered   
249646                 Unanswered   
258596  Black or African American   
261727                 Unanswered   
449852  Black or African American   
460324  Black or African American   
669928                 Unanswered   
715327                 Unanswered   
727495  Black or African American   
734709  Black or African American   
784812                 Unanswered   

                                             BorrowerName  \
19967   REORGANIZED SAINT JAMES TEMPLE AFRICAN METHODI...   
42228          TRINITY AFRICAN METHODIST EPISCOPAL CHURCH   
142067  PAYNE CHAPEL AFRICAN METHODIST EPISCOPAL CHURC...   
244018  BETHEL AFRICAN METHODIST EPISCOPAL CHURCH OF L...   
249603   ST. JOHN AFRICAN METHODIST EPISCOPAL CHURCH INC.   
249646  ST. JOHN AFRICAN METHODIST EPISCOPAL CHURCH, INC.   
258596           UNITY AFRICAN METHODIST EPISCOPAL CHURCH   
261727      MOUNT OLIVE AFRICAN METHODIST EPISOPAL CHURCH   
449852                  FIRST AFRICAN METHODIST EPISCOPAL   
460324                  FIRST AFRICAN METHODIST EPISCOPAL   
669928            FIRST AFRICAN METHODIST EPISOCPAL CHURC   
715327  UNIVERSITY AFRICAN METHODIST EPISCOPAL ZION CH...   
727495   BETHEL AFRICAN METHODIST EPISCOPAL CHURCH OXNARD   
734709   BETHEL AFRICAN METHODIST EPISCOPAL CHURCH OXNARD   
784812            WARD AFRICAN METHODIST EPISCOPAL CHURCH   

                               BusinessType  
19967               Non-Profit Organization  
42228               Non-Profit Organization  
142067              Non-Profit Organization  
244018              Non-Profit Organization  
249603  Employee Stock Ownership Plan(ESOP)  
249646  Employee Stock Ownership Plan(ESOP)  
258596              Non-Profit Organization  
261727              Non-Profit Organization  
449852                          Corporation  
460324                          Corporation  
669928              Non-Profit Organization  
715327                 501(c)3 – Non Profit  
727495                 501(c)3 – Non Profit  
734709                 501(c)3 – Non Profit  
784812              Non-Profit Organization

According to Frontline(https://www.pbs.org/wgbh/americanexperience/features/godinamerica-black-church/):Today "the black church" is widely understood to include the following seven major black Protestant denominations: the National Baptist Convention, the National Baptist Convention of America, the Progressive National Convention, the African Methodist Episcopal Church, the African Methodist Episcopal Zion Church, the Christian Methodist Episcopal Church and the Church of God in Christ.

So let's look for those seven denoms.

In [23]:
df[df['BorrowerName'].str.contains('NATIONAL BAPTIST | PROGRESSIVE NATIONAL | AFRICAN METHODIST | METHODIST EPISCOPAL | CHURCH OF GOD IN CHRIST')] \
    [['Race','BorrowerName','BusinessType','CurrentApprovalAmount','ForgivenessAmount','NAICSCode']]

Race  \
18274                         Unanswered   
18690                         Unanswered   
19967                         Unanswered   
42228                         Unanswered   
53596                         Unanswered   
102550         Black or African American   
104200         Black or African American   
105987  American Indian or Alaska Native   
106014                        Unanswered   
127226                        Unanswered   
128077                        Unanswered   
136479                        Unanswered   
141345                        Unanswered   
142067                        Unanswered   
142261                        Unanswered   
154864  American Indian or Alaska Native   
155275  American Indian or Alaska Native   
204179                        Unanswered   
224046         Black or African American   
226435                        Unanswered   
228306                        Unanswered   
244018                        Unanswered   
249603                        Unanswered   
249646                        Unanswered   
250166                        Unanswered   
255221         Black or African American   
258596         Black or African American   
261727                        Unanswered   
298500                        Unanswered   
350316                        Unanswered   
395253                        Unanswered   
449852         Black or African American   
454076                        Unanswered   
460324         Black or African American   
488802                        Unanswered   
500680                        Unanswered   
539605                        Unanswered   
544010                        Unanswered   
626854                        Unanswered   
669928                        Unanswered   
676905                        Unanswered   
685361                        Unanswered   
715327                        Unanswered   
727495         Black or African American   
734709         Black or African American   
741640                        Unanswered   
784812                        Unanswered   
813142                        Unanswered   
816859         Black or African American   

                                                        BorrowerName  \
18274               FIRST CHRISTIAN METHODIST EPISCOPAL CHURCH, INC.   
18690                FIRST CHRISTIAN METHODIST EPISCOPAL CHURCH INC.   
19967   REORGANIZED SAINT JAMES TEMPLE AFRICAN METHODIST EPISCOPAL Z   
42228                     TRINITY AFRICAN METHODIST EPISCOPAL CHURCH   
53596                            THOMASVILLE CHURCH OF GOD IN CHRIST   
102550           FOUNTAIN OF LIFE MINISTRIES CHURCH OF GOD IN CHRIST   
104200                         HARRIS TEMPLE CHURCH OF GOD IN CHRIST   
105987                          CAPITOL CITY CHURCH OF GOD IN CHRIST   
106014                          CAPITOL CITY CHURCH OF GOD IN CHRIST   
127226                     RESTORATION PLACE CHURCH OF GOD IN CHRIST   
128077                        GRACE COVENANT CHURCH OF GOD IN CHRIST   
136479                     GOSPEL TABERNACLE CHURCH OF GOD IN CHRIST   
141345                    NEW LIFE INTERNATIONAL BAPTIST CHURCH INC.   
142067  PAYNE CHAPEL AFRICAN METHODIST EPISCOPAL CHURCH OF BIRMINGHA   
142261          FERGUSON CHAPEL CHRISTIAN METHODIST EPISCOPAL CHURCH   
154864                      GRACE TABERNACLE CHURCH OF GOD IN CHRIST   
155275                      GRACE TABERNACLE CHURCH OF GOD IN CHRIST   
204179                                MARION CHURCH OF GOD IN CHRIST   
224046                         FIRST TRINITY CHURCH OF GOD IN CHRIST   
226435                    NEW CALVARY TEMPLE CHURCH OF GOD IN CHRIST   
228306                        MILLERS TEMPLE CHURCH OF GOD IN CHRIST   
244018  BETHEL AFRICAN METHODIST EPISCOPAL CHURCH OF LITTLE ROCK, AR   
249603              ST. JOHN AFRICAN METHODIST EPISCOPAL CHURCH INC.   
249646             ST. JOHN AFRICAN METHODIST EPISCOPAL CHURCH, INC.   
250166                           PR

This data shows that there is not a consistent pattern for when a BC provides a value for the "Race" race column. So we can't search for race to identify these groups.

It is interesting that ALL the matches in the 900K database show up here. This seems to be a rather short list -- how many entries are there?

In [37]:
len(df[df['BorrowerName'].str.contains('NATIONAL BAPTIST | PROGRESSIVE NATIONAL | AFRICAN METHODIST | METHODIST EPISCOPAL | CHURCH OF GOD IN CHRIST')] )

49

So of the 2,649 groups identified as religious organizations, only 49 were HBC's? That's 1.85% of the total number of ROs.

What percent of the 2020 national data did these groups comprise? Fortunately, 2020 had the survey of American religious congregrations. So at the same time the PPP loans were being disbursed, there was a survey to find information about religious congregrations. So let's get some of thaat data to see what percent of Am congreg belong to one of these seven traditions.

In [8]:
natl = pd.read_excel('/home/alp/Google Drive/Python/analysis/MappingReligion2020/2020 USRC Group Detail_0.xlsx',sheet_name = '2020 Group by Nation')

In [40]:
natl.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 374 entries, 0 to 373
Data columns (total 6 columns):
 #   Column                             Non-Null Count  Dtype  
---  ------                             --------------  -----  
 0   GROUP CODE                         373 non-null    object 
 1   GROUP NAME                         372 non-null    object 
 2   CONGREGATIONS                      373 non-null    float64
 3   ADHERENTS                          217 non-null    float64
 4   Adherents as % of Total Adherents  217 non-null    float64
 5   Adherents as % of Population       217 non-null    float64
dtypes: float64(4), object(2)
memory usage: 17.7+ KB


In [41]:
natl.head()

GROUP CODE                               GROUP NAME  CONGREGATIONS  \
0        326                 (Original) Church of God           15.0   
1        001                  Advent Christian Church          249.0   
2        003       African Methodist Episcopal Church         3667.0   
3        005  African Methodist Episcopal Zion Church         1587.0   
4        FHJ               Agape Christian Fellowship           21.0   

   ADHERENTS  Adherents as % of Total Adherents  Adherents as % of Population  
0        NaN                                NaN                           NaN  
1    17780.0                           0.000110                      0.000054  
2  1059888.0                           0.006568                      0.003198  
3   304704.0                           0.001888                      0.000919  
4     2616.0                           0.000016                      0.000008

In [42]:
natl.tail()

GROUP CODE                                  GROUP NAME  CONGREGATIONS  \
369        FD9          Wisler Mennonite Conference (Ohio)            5.0   
370        FE1  Wisler Mennonite Conference (Ohio-Indiana)           10.0   
371        490                                 Zoroastrian           33.0   
372        NaN                                         NaN            NaN   
373     Totals                                         NaN       356723.0   

       ADHERENTS  Adherents as % of Total Adherents  \
369        644.0                           0.000004   
370       1753.0                           0.000011   
371          NaN                                NaN   
372          NaN                                NaN   
373  161371931.0                           1.000000   

     Adherents as % of Population  
369                      0.000002  
370                      0.000005  
371                           NaN  
372                           NaN  
373                      0.486868

Drop the last two rows in the table (which don't have GROUP NAME values:

In [9]:
natl = natl.dropna(subset = ['GROUP NAME'])

According to this spreadsheet, there are 356,723 congs in the USA. How many are among the HBCs?

In [138]:
hbcs = 0
hbc_list = ['National Baptist','Progressive National','African Methodist Episcopal','Church of God in Christ']
print(f'{"Name":<50}{"N Cong":<10}')
for name in hbc_list:
    #print(name)

    for row in natl[natl['GROUP NAME'].str.contains(name)].iterrows():
        print(f'{row[1][1]:<50} {row[1][2]:<10}') #.row[1][1],row[1][2] )
        hbcs += row[1][2]

print(f'Total number of cong in the BC family {hbcs}.')

Name                                              N Cong    
National Baptist Convention of America, Inc.       389.0     
National Baptist Convention, USA, Inc.             2530.0    
Progressive National Baptist Convention, Inc.      363.0     
Progressive National Baptist Convention, Inc.      363.0     
African Methodist Episcopal Church                 3667.0    
African Methodist Episcopal Zion Church            1587.0    
Church of God in Christ                            3313.0    
Church of God in Christ, Mennonite                 170.0     
Total number of cong in the BC family 12382.0.


Why are PNBC double counted? Let's subtract one value and calculate the percent of the total N of cong:

In [83]:
(12382-363)/356723 * 100

3.369280926657378

So in the census of congregrations, the 7 BC families comprise about 3.37% of all congregrations in the USA.

Recall that in the PPO data, of the money disbursed to 2,638 (0.29% of the total PPO) religious organizations, 1.85% was disbursed to congs in the BC family.

There maybe innoculous explainations for this disparity (perhaps the BCs have smaller staffs or unpaid ministers) but at least part of the explanation must be the difficulty of applying for PPO funds. To do so required some familarity with the banking industry, legal knowledge helped, etc.

As a check on this, let's see what Episcopal Church and Jewish groups look like (these groups have higher-than-average education and income levels; see the PEW religious landscape survey data for information about this). What percent of the national group do they represent? Did they get this amount from the PPO pool or a higher number?

In [10]:
jewish = 0

print(f'{"Name":<50}{"N Cong":<10}')
for row in natl[natl['GROUP NAME'].str.contains('Judaism')].iterrows():
        print(f'{row[1][1]:<50} {row[1][2]:<10}') #.row[1][1],row[1][2] )
        jewish += row[1][2]

print(f'Total number of cong in the Jewish family {jewish}.')

Name                                              N Cong    
Chabad Judaism                                     953.0     
Conservative Judaism                               529.0     
Independent Judaism                                130.0     
Orthodox Judaism                                   914.0     
Reconstructionist Judaism                          90.0      
Reform Judaism                                     690.0     
Total number of cong in the Jewish family 3306.0.


So there are 3,306 Jewish congs in the cong survey data --> 0.925% of all congs.

There are 6,353 Episcopal Church congs (this is easy to read from the spreadsheet). 
This is 1.781% of all congs.

Now, how many relig org (total: 2,638) PPO loans went to each of these organizations?


In [17]:
df[(df['BorrowerName'].str.contains('SYNAGOGUE | JEWISH | ISRAEL')) & (df['BusinessType'] == 'Non-Profit Organization')]


LoanNumber DateApproved  SBAOfficeCode ProcessingMethod  \
11252   7293288303   01/28/2021         1084.0              PPS   
12695   8522977203   04/28/2020         1084.0              PPP   
21226   1761037209   04/15/2020          459.0              PPP   
21227   4533898403   02/06/2021          459.0              PPS   
21510   7412607006   04/07/2020          459.0              PPP   
...            ...          ...            ...              ...   
813260  5547557307   04/30/2020          914.0              PPP   
816045  2955067907   06/12/2020          931.0              PPP   
823200  5016457302   04/30/2020          914.0              PPP   
832804  7061108503   03/05/2021          912.0              PPS   
836586  8325788409   02/13/2021          914.0              PPS   

                                                        BorrowerName  \
11252                                ALASKA JEWISH HISTORICAL MUSEUM   
12695            ALASKA JEWISH HISTORICAL MUSEUM AND CULTURAL CENTER   
21226                                    N E MILES JEWISH DAY SCHOOL   
21227                                     NE MILES JEWISH DAY SCHOOL   
21510                                   BIRMINGHAM JEWISH FEDERATION   
...                                                              ...   
813260            TEMPLE BETH ISRAEL OF HIGHLAND PARK AND EAGLE ROCK   
816045  MIKVAH TAHARAS ISRAEL CONGREDGATION DBA VAAD OF NORTHERN CAL   
823200                       UNION OF MESSIANIC JEWISH CONGREGATIONS   
832804                           SILICON VALLEY JEWISH FILM FESTIVAL   
836586                       UNION OF MESSIANIC JEWISH CONGREGATIONS   

                             BorrowerAddress      BorrowerCity BorrowerState  \
11252                        1117 E 35th Ave         Anchorage            AK   
12695                        1117 E 35th Ave         ANCHORAGE            AK   
21226                      4000 MONTCLAIR RD        BIRMINGHAM            AL   
21227                      4000 Montclair Rd        Birmingham            AL   
21510                      3966 MONTCLAIR RD      MOUNTAIN BRK            AL   
...                                      ...               ...           ...   
813260                   5711 MONTE VISTA ST       LOS ANGELES            CA   
816045                   3316. Mayfair Drive        Sacramento            CA   
823200  3717 E. Thousand Oaks Blvd Suite 305  Westlake Village            CA   
832804                     690 W Fremont Ave         Sunnyvale            CA   
836586     3717 E Thousand Oaks Blvd Ste 305  Westlake Village            CA   

       BorrowerZip LoanStatusDate    LoanStatus  Term  SBAGuarantyPercentage  \
11252   99508-4258     01/06/2022  Paid in Full    60                    100   
12695        99508     08/07/2021  Paid in Full    24                    100   
21226   35213-2412     02/18/2021  Paid in Full    24                    100   
21227   35213-2412     04/15/2022  Paid in Full    60                    100   
21510   35213-2413     06/05/2021  Paid in Full    24                    100   
...            ...            ...           ...   ...                    ...   
813260  90042-3427     11/19/2020  Paid in Full    24                    100   
816045  95864-3804     03/18/2022  Paid in Full    60                    100   
823200       91362     10/16/2021  Paid in Full    24                    100   
832804  94087-4200     10/08/2021  Paid in Full    60                    100   
836586  91362-3607     08/18/2021  Paid in Full    60                    100   

        InitialApprovalAmount  CurrentApprovalAmount  UndisbursedAmount  \
11252                20827.00               20827.00                0.0   
12695                17199.34               17199.34                0.0   
21226               149537.50              149537.50                0.0   
21227               149537.50              149537.50                0.0   
21510               146000.00            

In [19]:
df['BorrowerState'].value_counts()

CA    470222
AL    158648
AZ    153454
AR     96184
AK     21031
AS       304
AE         4
Name: BorrowerState, dtype: int64

In [20]:
# OK, so we need to get ALL the data, since it is sorted by state, not random or by date of loan application.
df['RuralUrbanIndicator'].value_counts()

U    735719
R    164279
Name: RuralUrbanIndicator, dtype: int64

In [22]:
df['BusinessAgeDescription'].value_counts(normalize=True)

Existing or more than 2 years old         0.873735
New Business or 2 years or less           0.085020
Unanswered                                0.040848
Startup, Loan Funds will Open Business    0.000260
Change of Ownership                       0.000137
Name: BusinessAgeDescription, dtype: float64